### DISCUSSION TITLE
Any updates on "Vertical Taskbar for Windows 11?"

### DISCUSSION THREAD

**Discussion by Sand216** [ramensoftware/windhawk-mods#882] [Q&A]
@m417z Just wondering if there is a new version planned for this mod incorporating the bug fixes and features shown in #730 and #756, as well as fixing some other issues it's 1.0 release has.

No pressure to work on anything, I just want to gauge if this mod could be a viable competitor to the StartAllBack vertical taskbar in the future. 🙂

**Comment by m417z** [comment]
Yes, a new version is planned, but I don't have a timeline to provide. Your feedback can help me prioritize my time on things that matter. Have you tried the mod? What bothered you the most? What was missing? Is it usable, or is something a deal-breaker for using it?

**Reply by Sand216** [reply]
I have tried the mod for a little while, but I can't fully commit to using it.

Here is a list of problems I have faced, from "deal breaker" to "mild inconvenience".

I hope this list proves useful to you, let me know if you have any questions.

**1. Start menu cut off by taskbar**

This has already been covered by #730, so there shouldn't be a need for me to elaborate.

**2. Clock and some icons are sideways**

I know the clock has been addressed, but some other icons like the microphone are also sideways.

![clockAndMic](https://github.com/user-attachments/assets/59ee10a7-0484-4e61-826e-aa2dc218948b)

**3. Frequently disappearing icons & bugged animations**

It's almost guaranteed for icons to disappear after opening their corresponding program, and the opening animation where the icon bounces is also not right.

https://github.com/user-attachments/assets/d2209004-64ff-41bb-bb2c-0ea488f5c319

**4. Tray icon flyouts still at bottom right corner**

This has already been covered by #756.

**5. Start context menu is very misaligned**

The context menu is nowhere near the start button.

https://github.com/user-attachments/assets/ecffc7f9-70b7-4742-b70f-6e88cdcadc25

**6. Cut off window previews.**

When you close a window using the window preview, it gets shifted/cut off so that it is difficult to close other windows without reopening the window preview flyout.

https://github.com/user-attachments/assets/60859b55-b48e-4784-a054-87b50ab7b10f

**7. Start button size increases when taskbar length increases**

Not sure if this is intentional or not because the other icons do not increase in size, and this behavior was not present in Windows 10's vertical taskbar.

https://github.com/user-attachments/assets/d365ae40-7f9e-483a-b1e5-b14f59fd949d

**8. Misaligned minimize animation**

When you minimize a program, it shrinks to what looks to be the position of the icon if the taskbar was on top, and not on the left.

https://github.com/user-attachments/assets/737dfe8d-ed27-4ec1-9393-89742f36d579

**9. Cutoff jumplists**

The jumplists when you right click a program's icon are cut off if the program is near the top of the screen.

https://github.com/user-attachments/assets/5457ce3f-53f2-4da3-b684-f7ee4866cbcd

**10. Hidden tray icon menu looks out of place**

The menu still flys in from the bottom, and is a bit far from the taskbar.

https://github.com/user-attachments/assets/434fbf1a-796d-4355-84f2-62a00ca9b592

**11. Start and show desktop buttons cannot be clicked from their respective corners**

If you move your cursor to the very top left corner of the screen, you cannot open the start menu, you have to move your cursor slightly to the right. Same goes for the show desktop button but on the lower left corner.

**Honorable Mentions**

These problems do not affect me, as I do not use these features, but some other people may like them.

1. You cannot move the taskbar to the right.
2. Taskbar autohide is completely broken.

**Reply by m417z** [reply]
Thank you for the detailed list and for taking the time to attach screen recordings, it really helps! I knew about some of them, but some are new to me.

**Reply by m417z** [reply]
I've just released version 1.1 with many fixes and improvements, please try it!

> 1. Start menu cut off by taskbar
> 2. Clock and some icons are sideways

Fixed.

> 3. Frequently disappearing icons & bugged animations

Hopefully fixed.

"the opening animation where the icon bounces is also not right" - what do you mean?

> 4. Tray icon flyouts still at bottom right corner
> 5. Start context menu is very misaligned

Fixed.

> 6. Cut off window previews.

Didn't look at it yet. I knew it can be buggy, but didn't notice that it's so easy to trigger.

> 7. Start button size increases when taskbar length increases

Not intentional. Fixed.

> 8. Misaligned minimize animation

Didn't look at it yet.

> 9. Cutoff jumplists

Didn't look at it yet. I moved them slightly down as they were too far above the click location, so it should help a tiny little bit.

> 10. Hidden tray icon menu looks out of place

I know about it and I spent too much time trying to make it work. It's located closer to the taskbar when there are more icons, but because I didn't find a way to get the actual width of the container, I placed it so that it'll be usable for all possible container widths. I might revisit it in the future,

There is another, more serious issue with tray icons: dragging from the overflow popup to the tray and vice versa doesn't work. One workaround is to configure tray icon visibility via settings, but it's much less convenient.

> 11. Start and show desktop buttons cannot be clicked from their respective corners

Didn't look at it yet.

> You cannot move the taskbar to the right.

On my list.

> Taskbar autohide is completely broken.

Didn't look at it yet.

Please let me know if anything that's supposed to be fixed is still a problem, or if there are any new problems.

**Reply by m417z** [reply]
I created an issue to make it easier to track this list: https://github.com/ramensoftware/windhawk-mods/issues/913.

**Reply by m417z** [reply]
> Taskbar autohide is completely broken.

I never tried it, and I was pretty sure it'd crash or screw up the taskbar beyond recognition. It wasn't that bad. It has a funny animation instead, and the taskbar is even usable some of the time, and seems to be back to normal when autohide is disabled.

https://github.com/user-attachments/assets/ce989f88-f172-41ef-935d-9ef549d8aa69

**Reply by Sand216** [reply]
> "the opening animation where the icon bounces is also not right" - what do you mean?

The new update actually makes this more apparent. To get an idea of what I am talking about, here is a recording of the default Windows 11 taskbar, flipped in the video. This video shows what the default, unmodified animations should look like.

https://github.com/user-attachments/assets/eddde875-63d7-4b6a-95bd-2379a931a132

The "bounce" I am talking about is the translation of the icon up and down (left and right for vertical taskbar) when a window is minimized, restored, opened, or closed.

With the mod on, these translations do not occur properly and the icon either doesn't move or moves incorrectly.

https://github.com/user-attachments/assets/aa860a75-bd49-4ff8-bf20-cf9fa1c64e95

In this example, the icon only translates when a window is restored, and nothing happens when minimized. It only "bounces" to the right.

https://github.com/user-attachments/assets/b53f3d1f-c382-4e95-82e2-090efa93d937

In this example, there are no translations at all, and only the scale effect can be seen. There are no "bounces".

Something similar also happens when opening a program that is not pinned to the taskbar. Normally it slides in from the bottom (left in this case) when opened and slides back out when closed, both using the same "bouncing" animation.

https://github.com/user-attachments/assets/45992b94-60d4-473f-a0cb-a6c4219183b0

In this example, the run icon correctly slides out, but incorrectly slides in.

**These are all minor details and no functionality is affected by these issues, it is only cosmetic.**

**Reply by Sand216** [reply]
I have been using the mod since the update today and it has been working great! Nice work. I'll keep you posted if I find any other quirks.

**Reply by m417z** [reply]
> With the mod on, these translations do not occur properly

Gotcha, thanks for the detailed explanation. I didn't touch these animations, so it looks like the rotation/position changes screw it up. Not sure how easy it'd be to fix, but doesn't look like a high priority, I'll look at it at some point. I added it to the issue too.

**Reply by Sand216** [reply]
Do you want me to continue posting here, or on the dedicated issue tracker post you made?

There is a new bug I have found regarding the start menu when using the Windows 11 Start Menu Styler mod. If the start menu is too long (e.x. the all apps list is shown along with pinned) it will get cut off.

![startMenuNormal](https://github.com/user-attachments/assets/4f5a5756-adbb-4b19-be5f-b715b51f7175)
My edited version of the "SideBySide2" start menu theme displaying normally with bottom taskbar.

![clippedStartMenu](https://github.com/user-attachments/assets/8232f941-c9ad-48eb-9f0b-ea268a584a41)
That same theme clipped at the right side with vertical taskbar.

![theOldSignOutMenuWasBetter](https://github.com/user-attachments/assets/c7c7f40f-ae47-4362-a547-a593fe1b8b06)
Also, the new sign out + Microsoft advertisement thingy gets cut off. The old sign out menu was so much better...

**Reply by Sand216** [reply]
The multi window element that appears when you have more than one window open with grouped taskbar buttons doesn't stretch across the whole button. You could maybe flip it and put it on the right side of the button or find a way to stretch it to keep it on the bottom. Default win11 taskbar has it at the right though.

![image](https://github.com/user-attachments/assets/265c9a60-a8e3-47e4-9ad2-a35eb8707238)

**Reply by m417z** [reply]
> Do you want me to continue posting here, or on the dedicated issue tracker post you made?

Here is fine.

> There is a new bug I have found regarding the start menu when using the Windows 11 Start Menu Styler mod. If the start menu is too long (e.x. the all apps list is shown along with pinned) it will get cut off.

Yes, I knew it'd be a problem. It's tricky to solve, though. What I could do fairly easily is to add a configuration to settings to control the start width, then it could be adjusted manually. Not ideal but better than nothing.

Meanwhile, you can try it to fork the mod and adjust it yourself here (currently 660):
https://github.com/ramensoftware/windhawk-mods/blob/ebdfc3b11e6dedeb63bf66dd75ed00cc7a8ceb47/mods/taskbar-vertical.wh.cpp#L1587

> The multi window element [...] doesn't stretch across the whole button

Yes, saw that too, I'll add it to the list, thanks.

**Reply by Sand216** [reply]
> Meanwhile, you can try it to fork the mod and adjust it yourself here (currently 660):

Thanks, I changed it to 775 and it is not cut off anymore.

**Reply by m417z** [reply]
Done/fixed in v1.2:
* Cut off window previews
* You cannot move the taskbar to the right
* If the start menu is too long (e.g. with Start Menu Styler) it will get cut off - added an option to customize the width

**Reply by m417z** [reply]
BTW I couldn't reproduce this one:

> 11. Start and show desktop buttons cannot be clicked from their respective corners

**Reply by Sand216** [reply]
> BTW I couldn't reproduce this one

Here's a screen recording. By corners, I mean the complete upper/lower left corner, like the farthest in the corner the mouse can go.

https://github.com/user-attachments/assets/89153736-cd0b-4f8c-b608-920d3a3fe985

**Reply by m417z** [reply]
Somebody reported on Reddit that the issue happens for some widths, but not for others:
https://www.reddit.com/r/Windows11/comments/1fktwn6/comment/lo7mmmk/
> My only issue is at certain taskbar width on coordinate (0,0) I cannot press start button. the entire (0) column does not register clicks. I need to be at least in column (1) for button clicks to register. This is true at width 60 but forks fine at width 58 for example.

**Reply by Sand216** [reply]
> Somebody reported on Reddit that the issue happens for some widths, but not for others:

Interesting

Also, the latest mod update seems to have broken restarting explorer via a non-administrator command prompt. Explorer restarts, but the taskbar does not appear. This issue was not present in prior versions, and does not apply to administrator cmd windows or task manager.


https://github.com/user-attachments/assets/e99852e3-8b86-4dd2-8e04-c7173009a1a9

**Reply by m417z** [reply]
Both issues (the corners and the process restart) should be fixed in version 1.2.3.

**Reply by ephedrin3** [reply]
When i use taskbar themes notification area is black and when i use normal skin in left is a orange line on top of the bar, so i use translucent. on right there is no line but still notification is missing. hope it will fix next update. thank you.

**Reply by m417z** [reply]
Vertical taskbar styling might not work depending on the theme. I haven't looked at it, not sure if the fix should be on the theme side or the vertical taskbar mod side. You can see a relevant discussion here:
https://github.com/ramensoftware/windows-11-taskbar-styling-guide/issues/87

**Reply by m417z** [reply]
> Hidden tray icon menu looks out of place

The position should be fixed in 1.2.5. The animation, well, it's all over the place, including jump lists and the start menu. Not a priority.

**Reply by Sand216** [reply]
Regarding the jump lists, ExplorerPatcher implements some black magic to make them work and look correct. Not sure if it could be helpful to you. No pressure, I understand this is not a priority.

https://github.com/user-attachments/assets/fa006425-d3fe-43af-aef0-14b5dcf9af14

**Reply by m417z** [reply]
Thanks, I see they're doing some patching here:
https://github.com/valinet/ExplorerPatcher/blob/92fce8c90267fc9036e816435badc71b12027380/ExplorerPatcher/TwinUIPatches.cpp#L2669

The `CJumpViewExperienceManager::EnsureWindowPosition` function is a good lead, I'll look into it. But aside from the problem with items on top of the monitor, I liked the fact that the close button is always right where the cursor is. Do you think the behavior on your recording is more preferable?

**Reply by Sand216** [reply]
In my opinion, yes it is preferable, It's how it was in Windows 10. 

![image](https://github.com/user-attachments/assets/a53f91bc-662c-4d18-b76e-c8fb71b11103)

I think having the menu not cut off is more important than saving some mouse movement. I can see the benefit of being able to quickly close programs though, is it too niche to make a mod option?

**Reply by m417z** [reply]
I have an initial implementation that opens it in the center, you can get it here to try it:
https://github.com/m417z/my-windhawk-mods/blob/6fa73e083014bd35536424153c8e91adaec1b1fc/mods/taskbar-vertical.wh.cpp

**Reply by Sand216** [reply]
Seems to be working pretty well! Having it in the center is a good compromise between having the close button near the mouse and not having the menu cut off. Does the EP code give any hints on how to fix the animations, once the high priority stuff is out of the way?

**Reply by m417z** [reply]
I was focused on the position, didn't notice they're handling the animation too :) yeah, it will likely help.
